Failed to read the public key content from 'mp.jwt.verify.publickey.location' in native with latest Quarkus main

### Describe the bug

Failed to read the public key content from 'mp.jwt.verify.publickey.location' in native with latest Quarkus main. Test works fine in JVM mode.

Works fine with Quarkus 3.21.0 in native, so it must be recent change in main.

```
2025-03-21 11:22:17,755 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-eventloop-thread-3) HTTP Request to /hello failed, error id: 6968d40d-6ed6-49e5-b63d-06dfc462a73b-1: jakarta.enterprise.inject.spi.DeploymentException: SRJWT02002: Failed to read the public key content from 'mp.jwt.verify.publickey.location'
	at io.smallrye.jwt.config.JWTAuthContextInfoProvider.getOptionalContextInfo(JWTAuthContextInfoProvider.java:734)
	at io.smallrye.jwt.config.JWTAuthContextInfoProvider.getContextInfo(JWTAuthContextInfoProvider.java:920)
	at io.smallrye.jwt.config.JWTAuthContextInfoProvider_ProducerMethod_getContextInfo_IdERZ3rgTvGnz5ERAK80guTDswo_Bean.doCreate(Unknown Source)
	at io.smallrye.jwt.config.JWTAuthContextInfoProvider_ProducerMethod_getContextInfo_IdERZ3rgTvGnz5ERAK80guTDswo_Bean.create(Unknown Source)
	at io.smallrye.jwt.config.JWTAuthContextInfoProvider_ProducerMethod_getContextInfo_IdERZ3rgTvGnz5ERAK80guTDswo_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.c85(Unknown Source)
	at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
...
Caused by: java.net.MalformedURLException: no protocol: /foo/bar
	at java.base@21.0.5/java.net.URL.<init>(URL.java:772)
	at java.base@21.0.5/java.net.URL.<init>(URL.java:654)
	at java.base@21.0.5/java.net.URL.<init>(URL.java:590)
	at io.smallrye.jwt.util.ResourceUtils$UrlStreamResolver.resolve(ResourceUtils.java:122)
	at io.smallrye.jwt.util.ResourceUtils.getResourceStream(ResourceUtils.java:82)
	at io.smallrye.jwt.util.ResourceUtils.readResource(ResourceUtils.java:45)
	at io.smallrye.jwt.util.ResourceUtils.readResource(ResourceUtils.java:40)
	at io.smallrye.jwt.config.JWTAuthContextInfoProvider.getOptionalContextInfo(JWTAuthContextInfoProvider.java:729)
	... 125 more
```

`mp.jwt.verify.publickey.location=/foo/bar` is defined in https://github.com/quarkus-qe/beefy-scenarios/blob/main/003-quarkus-many-extensions/src/main/resources/application.properties to get over the check for definition presence, but jwt is not really used

### Expected behavior

Both JVM and Native work, tests pass.

### Actual behavior

JVM works, tests pass. Native doesn't work, fails on firs request.

### How to Reproduce?

Setup Mandrel or GraalVM

 * Use https://github.com/quarkus-qe/beefy-scenarios
 * Run `mvn clean verify -Dnative -f 003-quarkus-many-extensions`

`mvn clean verify -Dnative -f 003-quarkus-many-extensions -Dquarkus.platform.version=3.21.0` works fine



### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

_No response_

### Mandrel or GraalVM version (if different from Java)

MANDREL 23.1.5.0 JDK 21.0.5+11-LTS

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/smallrye-jwt/deployment/src/main/java/io/quarkus/smallrye/jwt/deployment/SmallryeJwtDevModeProcessor.java
+++ b/extensions/smallrye-jwt/deployment/src/main/java/io/quarkus/smallrye/jwt/deployment/SmallryeJwtDevModeProcessor.java
@@ -112,7 +112,7 @@ public class SmallryeJwtDevModeProcessor {
             devServiceConfigs.put(SMALLRYE_JWT_SIGN_KEY, NONE);
         }
 
-        if (!devServiceConfigs.containsKey(MP_JWT_VERIFY_PUBLIC_KEY)) {
+        if (!userConfigs.contains(MP_JWT_VERIFY_PUBLIC_KEY)) {
             devServiceConfigs.put(MP_JWT_VERIFY_PUBLIC_KEY, NONE);
         }
 

